src/Generator/SchemaMapperGenerator.php

```diff
@@ -374,25 +388,59 @@ protected function generateMapStatementsForObject(Field $root, Variable $payload
                     $optionalVar = $optionalResponseItems[$i];
                 }
 
-                if ($field->isNullable()) {
-                    $ifCondition = $this->builder->funcCall(
-                        'array_key_exists',
-                        [
-                            $field->getName(),
-                            $payloadVariable,
-                        ]
+                if (!$root->hasOneOf()) {
```

Comment:
same comment I made on the other PR, I think this code is modelling the anyOf, not the oneOf (the difference being oneOf must match only 1 schema)

Reply:
we could add support for both anyOf and oneOf here IMO, and just do the extra check/constraint about the matching 1,N schemas, wdyt?